src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsProjectJsonToPackageReferenceMigrator.cs

```diff
@@ -41,16 +39,12 @@ public IVsProjectJsonToPackageReferenceMigrateResult MigrateProjectJsonToPackage
                 throw new FileNotFoundException(nameof(projectUniqueName));
             }
 
-            return  NuGetUIThreadHelper.JoinableTaskFactory.Run(async () => 
-            {
-                await TaskScheduler.Default;
-                return await MigrateProjectToPackageRefAsync(projectUniqueName);
-            });
-            
+            return await MigrateProjectToPackageRefAsync(projectUniqueName);            
         }
 
         private async Task<IVsProjectJsonToPackageReferenceMigrateResult> MigrateProjectToPackageRefAsync(string projectUniqueName)
-        {            
+        {
+            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
Just saw your whole method and realized why you need to run the whole method on UI thread? You shouldn't switch to UI thread here at all. All these _solutionManager API calls take care of switching to UI thread whenever needed